mathesar/views/api.py

```diff
@@ -85,11 +85,13 @@ def create(self, request):
         serializer.is_valid(raise_exception=True)
 
         valid_data = serializer.validated_data
-        if 'data_files' in valid_data and valid_data['data_files']:
-            table = create_table_from_datafile(valid_data)
-        else:
-            table = create_empty_table(valid_data)
-
+        try:
+            if 'data_files' in valid_data and valid_data['data_files']:
+                table = create_table_from_datafile(valid_data)
+            else:
+                table = create_empty_table(valid_data)
+        except InvalidRequestError:
```

Comment:
There might be other reasons for an `InvalidRequestError`, I would double check that it's because the table already exists. For an example of handling a more specific error, see how `DuplicateColumn` is handled in this file.

Reply:
Excellent point, I will make this more precise,

Reply:
@kgodey i adjusted the check to be more precise, but in doing so needed to remove the `.reflect` call in the underlying function. Do you see any issues with doing that?

When using reflect, sqlacademy handles the exception and wraps it in it's own `InvalidRequestError` and doesn't seem to expose any extra detail beyond a string of what the error is, which gives some unhelpful advice(`sqlalchemy.exc.InvalidRequestError: Table 'public.blerp' is already defined for this MetaData instance.  Specify 'extend_existing=True' to redefine options and columns on an existing Table object.`)

Reply:
I don't see any issues with it, tagging @mathemancer to double check since he wrote that code.